### DISCUSSION TITLE
CRUD for tokens

### DISCUSSION THREAD

**Discussion by pottekkat** [meshery/meshery#3406] [Ideas]
Users can have multiple tokens and reference them in their meshconfig (`~/.meshery/config.yaml`) file. This means that a user can have multiple tokens and can use different tokens for different contexts.

However, right now the users have to manually edit their config files in-order to change the token used in a particular context and even add new tokens, change the location of the token etc.

Similar to channel and context constructs in the meshconfig, mesheryctl should also support CRUD actions for tokens:

Add a new token to the meshconfig:

- `mesheryctl system token create <token name> -f <optional: path to the token, defaults to auth.json>` 
- Better verb than "create"? as create could mean create a new token.

Delete token:

- `mesheryctl system token delete <token name>`

Set token:

- `mesheryctl system token set <token name>`

List tokens:

- `mesheryctl system token list`

Power/advanced users can directly edit the meshconfig for use cases other than these.

I think these ought to cover all the common use cases. Please suggest any changes or updates needed here.

**Comment by piyushsingariya** [comment]
What about "add" instead of "create"?

**Reply by pottekkat** [reply]
"create" was only preferred for consistency. "add" is more meaningful. It is a trade-off I guess.
